### DISCUSSION TITLE
4 settembre 2023 - esercizio 1

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#44] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/498f6400-c42a-486d-8fec-164ea44451dc)

**Comment by IonutCicio** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/0c9533d9-ea66-4631-ba6e-b1c96037dd10)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/ef87dd0e-3a0b-431d-939f-d9fa2fbee80e)

**Comment by Warcophyr** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/e0feae27-4655-451d-817f-87b07226ce5d)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/54fdc8d5-fa0b-4beb-9a95-bda4efbf8a6d)

**Comment by Elia-Belli** [comment]
**(1)**
- Progetti più vecchi di 3 anni: $Prog= \sigma_{(DataI\lt 01-09-2020)}Progetto$
- Dipendenti desiderati : $Dip=\pi_{(DipenId)}((Prog\bowtie_{(Id=ProgId)}(\sigma_{(Ruolo=Designer\ \lor\ Ruolo=Programmer)}Team)))$
- Risposta: $\pi_{(Nome,Cognome,DataN)}(Dip\bowtie_{(DipenId=Id)}Dipendente)$

**(2)**
- Copia Progetto: $ProgettoC= \rho_{(AC\leftarrow A,\forall A\in Progetto )}Progetto$
- Progetti più recenti di almeno un altro progetto: $Prog\\_ Rec = \pi_{(Id,SettId,Titolo,DataI)}(Progetto\bowtie_{(DataI\gt DataIC)}ProgettoC)$
- Progetti più vecchi: $ProgV=Progetto-Prog\\_ Rec$
- Progetti in cui qualche dipendente non è stato Project Leader: $NotAll\\_ PL= (\sigma_{(Ruolo!=ProjectLeader)}Team)\bowtie_{(ProgId=Id)}Progetto$
- Progetti più vecchi in cui ogni dipendente è stato ProjectLeader: $ProgV\\_ PL= ProgV-NotAll\\_ PL$
- Risposta: $\pi_{(Nome,Cognome, Titolo)}((ProgV\\_ PL \bowtie_{(Id=ProgId)} Team)\bowtie_{(DipenId=Id)}Dipendente)$

**Comment by CiottoloMaggico** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/79638284/6d37b057-ab14-4edf-9c94-df31f229996a)

**Comment by Frisbon** [comment]
### Secondo voi l'implementazione é giusta?
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/84731341/9d46bba7-a1e1-4774-88b6-14a42119d38d)

**Comment by daromacarlo** [comment]
![Immagine 2024-02-11 194135](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/116788f3-ff6e-4c2b-9db9-5f94310be478)
![Immagine 2024-02-11 194004](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/4ec58edb-796e-489d-93ab-07dd283cd12a)
